cli/pkg/kctrl/local/sources/git.go

```diff
@@ -11,62 +11,61 @@ import (
 	vendirconf "github.com/vmware-tanzu/carvel-vendir/pkg/vendir/config"
 )
 
-type GitStep struct {
-	ui           cmdcore.AuthoringUI
-	vendirConfig vendirconf.Config
+type SourceConfiguration interface {
+	Configure() error
 }
 
-func NewGitStep(ui cmdcore.AuthoringUI, vendirConfig vendirconf.Config) *GitStep {
-	return &GitStep{
-		ui:           ui,
-		vendirConfig: vendirConfig,
-	}
+type GitConfiguration struct {
+	ui           cmdcore.AuthoringUI
+	vendirConfig *VendirConfig
 }
 
-func (g *GitStep) PreInteract() error { return nil }
+var _ SourceConfiguration = &GitConfiguration{}
 
-func (g *GitStep) Interact() error {
-	contents := g.vendirConfig.Directories[0].Contents
+func NewGitConfiguration(ui cmdcore.AuthoringUI, vendirConfig *VendirConfig) *GitConfiguration {
+	return &GitConfiguration{ui: ui, vendirConfig: vendirConfig}
+}
+
+func (g *GitConfiguration) Configure() error {
+	contents := g.vendirConfig.Contents()
 	if contents == nil {
-		err := g.initializeContentWithGit()
+		err := g.initializeContentWithGit(contents)
 		if err != nil {
 			return err
 		}
 	} else if contents[0].Git == nil {
```

Comment:
Should there be a check on len of contents?

Reply:
That would mean the user removed the item added by kctrl
But yep makes sense I guess, will do these split up changes tomorrow